pkg/controller/spinnakerservice/status.go

```diff
@@ -3,62 +3,69 @@ package spinnakerservice
 	appsv1 "k8s.io/api/apps/v1beta2"
-
-	// corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/selection"
-
-	spinnakerv1alpha1 "github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type statusChecker struct {
 	client client.Client
 }
 
+const (
+	Ok          = "OK"
+	Updating    = "Updating"
+	Unavailable = "Unavailable"
+	Na          = "N/A"
+)
+
 func newStatusChecker(client client.Client) statusChecker {
 	return statusChecker{client: client}
 }
 
-type labelSelector struct{}
-
-func (l *labelSelector) Matches(labels labels.Labels) bool {
-	return labels.Get("app.kubernetes.io/managed-by") == "halyard"
-}
-
 func (s *statusChecker) checks(instance spinnakerv1alpha1.SpinnakerServiceInterface) error {
-	r, err := labels.NewRequirement("app.kubernetes.io/managed-by", selection.Equals, []string{"halyard"})
-	if err != nil {
-		return err
-	}
-	sel := labels.NewSelector()
-	sel.Add(*r)
 	// Get current deployment owned by the service
 	list := &appsv1.DeploymentList{}
-	err = s.client.List(context.TODO(), &client.ListOptions{LabelSelector: sel, Namespace: instance.GetNamespace()}, list)
+	err := s.client.List(context.TODO(), list, client.InNamespace(instance.GetNamespace()), client.MatchingLabels{"app.kubernetes.io/managed-by": "halyard"})
 	if err != nil {
 		return err
 	}
 
 	svcs := make([]spinnakerv1alpha1.SpinnakerDeploymentStatus, 0)
-	for i := range list.Items {
-		it := list.Items[i]
-		st := spinnakerv1alpha1.SpinnakerDeploymentStatus{
-			Name:                it.ObjectMeta.Name,
-			ObservedGeneration:  it.Status.ObservedGeneration,
-			Replicas:            it.Status.Replicas,
-			UpdatedReplicas:     it.Status.UpdatedReplicas,
-			ReadyReplicas:       it.Status.ReadyReplicas,
-			AvailableReplicas:   it.Status.AvailableReplicas,
-			UnavailableReplicas: it.Status.UnavailableReplicas,
-			LastUpdateTime:      it.ObjectMeta.CreationTimestamp,
-		}
-		svcs = append(svcs, st)
-	}
 	svc := instance.DeepCopyInterface()
 	status := svc.GetStatus()
-	status.Services = svcs
+	if len(list.Items) == 0 {
+		status.Status = Na
+		status.Services = []spinnakerv1alpha1.SpinnakerDeploymentStatus{}
+	} else {
+		status.Status = Ok
+		for i := range list.Items {
+			it := list.Items[i]
+
+			st := spinnakerv1alpha1.SpinnakerDeploymentStatus{
+				Name:                it.ObjectMeta.Name,
+				ObservedGeneration:  it.Status.ObservedGeneration,
+				Replicas:            it.Status.Replicas,
+				UpdatedReplicas:     it.Status.UpdatedReplicas,
+				ReadyReplicas:       it.Status.ReadyReplicas,
+				AvailableReplicas:   it.Status.AvailableReplicas,
+				UnavailableReplicas: it.Status.UnavailableReplicas,
+				LastUpdateTime:      it.ObjectMeta.CreationTimestamp,
+			}
+			// Spinnaker not ready if any of the services has zero pods
+			if it.Status.ReadyReplicas == 0 {
+				status.Status = Unavailable
```

Comment:
I'm curious what's the `ReadyReplicas` value when a deployment is starting. Let's say clouddriver pod is starting and doing a `kubectl get pods` shows that it has `0/1` pods ready. Doesn't this mean that Spinnaker is in status `Updating` instead of status `Unavailable`?
I just want to make sure that status `Unavailable` is only used when one or more spinnaker service deployments are down, but not while a new pod is being started while the old version is still running.

Reply:
Correct. The next statement should make it `unavailable` though since you'll have at least one updated replica that wouldn't be ready. So overall it would be `updating`.